C++/include/algorithm/dynamic_programming/longest_decreasing_subsequence.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef LONGEST_DECREASING_SUBSEQUENCE
+#define LONGEST_DECREASING_SUBSEQUENCE
+
+/*
+Longest Decreasing Subsequence
+---------------------
+This algorithm finds the longest decreasing  subsequence in a sequence of numbers
+It follows the principals of dynamic programming
+
+Time complexity
+----------------
+O(n logn) where n is the number of elements in the initial list
+
+Space complexity
+---------------
+O(n)
+
+*/
+
+#include <iostream>
+#include <vector>
+
+int longest_decreasing_subsequence(std::vector<int>& input,  const bool to_show_state = true) {//order of 1 means increasing subsequence, -1 for decreasing subsequence
+    int elements = input.size();//number of elements in the vector
```

Comment:
Please leave a white space between semicolon and inline comment.

`int example = 1; // like this`